### DISCUSSION TITLE
About dependency required installing libtensorflow_cc from conda channel.

### DISCUSSION THREAD

**Discussion by Cloudac7** [deepmodeling/deepmd-kit#1108] [General]
Well, it is a dirty trick to just jump the suffering step of compiling TensorFlow C++ interface from installing it from `conda`. I have used the very way to simplify installation steps for a lot of times, since there might be some protobuf issue using `conda` to install the whole DeePMD-kit environment for compiling some other interface from it.

However, when I try installation of `libtensorflow_cc=2.5.0=gpu_cuda10.1_0` (latest TensorFlow interface version providing support to CUDA 10.1), it raised `libiomp5.so` is lost without loading **Intel MPI** environmet as well as **Intel MKL**, which is not mentioned when using the former compiled 2.3.0 version. It seems also not installed from conda.

So is there any change in dependency of `libtensorflow_cc` from 2.3.0 or earlier version to 2.5.0, and should it be added for conda? And would it be OK to supply `libtensorflow_cc` release packages?

By the way, there might be something wrong using `libtensorflow_cc=2.5.0` compiling Lammps, raising some namespace error for eigen. More detail error info would add later. Rolling back to 2.3.0 may solve the issue, however for newer Nvidia Ampere Architecture devices, TensorFlow 2.3.0 might cause some issue and I have to upgrade to verison later than 2.4.1 (CUDA>=11.1).

**Comment by njzjz** [comment]
libomp5 is provided by [intel-openmp](https://anaconda.org/anaconda/intel-openmp).
